sail/src/main/java/org/apache/rya/rdftriplestore/inference/InferenceEngine.java

```diff
@@ -73,6 +74,8 @@
     private Set<URI> transitivePropertySet;
     private Map<Resource, Map<URI, Value>> hasValueByType;
     private Map<URI, Map<Resource, Value>> hasValueByProperty;
+    private Map<URI, Set<URI>> domainByType;
```

Comment:
We should probably start using ConcurrentHashMaps for all these maps.  The InferenceEngine is not very thread safe right now.